workflows/differentialabundance.nf

```diff
@@ -168,286 +47,518 @@ include { DIFFERENTIAL_FUNCTIONAL_ENRICHMENT                } from '../subworkfl
+            matrix: [meta, matrix]
+            samplesheet: [meta, samplesheet]
+        }
 
     CUSTOM_MATRIXFILTER(
-        ch_matrix_for_differential,
-        VALIDATOR.out.sample_meta
+        ch_matrixfilter_input.matrix,
+        ch_matrixfilter_input.samplesheet
     )
 
+    ch_filtered_matrix = prepareModuleOutput(CUSTOM_MATRIXFILTER.out.filtered, ch_paramsets)
+
     // ========================================================================
     // Differential analysis
     // ========================================================================
 
-    // Prepare inputs for differential processes
-
-    ch_differential_input = CUSTOM_MATRIXFILTER.out.filtered
-        .combine(ch_tools)
-        .map { meta, matrix, tools_norm, tools_diff, tools_func ->
-            [
-                meta,
-                matrix,
-                tools_diff.method,
-                tools_diff.fc_threshold,
-                tools_diff.stat_threshold
-            ]
+    ch_differential_input = ch_filtered_matrix
+        .join(ch_validated_samplemeta)
+        .join(ch_transcript_lengths)
+        .join(ch_control_features)
+        .join(ch_contrasts)
+
+    // Use a multiMap to generate synched channels for differential analysis
+    ch_differential_input = prepareModuleInput(ch_differential_input, 'differential')
+        .multiMap{ meta, matrix, samplesheet, transcript_lengths, control_features, contrast, variable, reference, target, formula, comparison ->
+            input: [meta, matrix, meta.params.differential_method, meta.params.differential_min_fold_change, meta.params.differential_max_qval]
+            samplesheet: [meta, samplesheet]
+            transcript_lengths: [meta, transcript_lengths]
+            control_features: [meta, control_features]
+            contrast: [meta, contrast, variable, reference, target, formula, comparison]
         }
 
     // Run differential analysis
 
     ABUNDANCE_DIFFERENTIAL_FILTER(
-        ch_differential_input,
-        VALIDATOR.out.sample_meta,
-        ch_transcript_lengths,
-        ch_control_features,
-        ch_contrasts
+        ch_differential_input.input,
+        ch_differential_input.samplesheet,
+        ch_differential_input.transcript_lengths,
+        ch_differential_input.control_features,
+        ch_differential_input.contrast
     )
 
-    // collect differential results
+    // Collect differential results
+    // note that 'differential_method' is additionally added to the meta in the differential subworkflow
+    // remove it to keep a consistent meta structure
+
+    // Also note that these channels, the meta contain other info apart from the base paramset meta
+    // hence we create a key using the simple paramset meta with only meta.analysis_name and meta.params
+    // this will facilitate later on to join/combine channels
+    // To keep nomenclature consistent with the rest of the pipeline, the simple meta is called as meta, whereas
```

Comment:
I'm finding the whole meta/ meta_full thing very confusing. Can we call `meta` `meta_params` or something?